### PR TITLE
WIP: Fix wikilinks slug querying

### PR DESCRIPTION
**Pull request by SanketDG**
This attempts to fix the wikilink generation issues.

### CODE REVIEW

File: packages/foam-core/src/note-graph.ts [hunk 1]
```diff
@@ -91,7 +91,7 @@ export class NoteGraph implements NoteGraphAPI {
     // prettier-ignore
     const filterFn =
       query == null ? (note: Note | null) => note != null
-        : 'slug' in query ? (note: Note | null) => note?.slug === query.slug
+        : 'slug' in query ? (note: Note | null) => [nameToSlug(query.slug), query.slug].includes(note?.slug as string)
```
Comment (SanketDG) [inline packages/foam-core/src/note-graph.ts:94]: I am checking for both [[Slug Name]] and [[slug-name]] here, however they are defined in the note, so both can be used interchangably.

Comment (SanketDG) [reply packages/foam-core/src/note-graph.ts:94]: @riccardoferretti

Comment (riccardoferretti) [reply packages/foam-core/src/note-graph.ts:94]: @SanketDG I have a feeling this wouldn't work with the dataviz graph, have you tested it?

Comment (SanketDG) [reply packages/foam-core/src/note-graph.ts:94]: Yes it definitely works with my notes, which all have filenames like "javascript-execution.md" and referenced as [[Javascript Execution]], this is uniform throughout.

Any other cases you want me to check?

**Review by riccardoferretti — APPROVED** [review]
LGTM!

**Pull request merged**